### PR TITLE
Add exception handler description in Spring web docs

### PR DESCRIPTION
**Pull request by hantsy**
Mentioned in issue #4042, added sections for exception handlers. This PR should be clean finally when #4042  and #5374 are resolved.

### CODE REVIEW

File: docs/src/main/asciidoc/spring-web.adoc [hunk 1]
```diff
@@ -233,12 +233,30 @@ The following method return types are supported:
 * POJO classes which will be serialized via JSON
 * `org.springframework.http.ResponseEntity`
 
-=== Controller method return types
+=== Controller method parameter types
```
Comment (geoand) [inline docs/src/main/asciidoc/spring-web.adoc]: Nice catch!

**Review by geoand — APPROVED** [review]
Thanks a lot!

**Comment by geoand** [comment]
Can you please squash into a single commit? Best not to have superfluous commits in the history

**Comment by cescoffier** [comment]
@geoand should it be backported?

**Comment by geoand** [comment]
@cescoffier yes yes!

**Comment by cescoffier** [comment]
I've restarted the failing tests as they look unrelated.

File: docs/src/main/asciidoc/spring-web.adoc [hunk 2]
```diff
@@ -233,12 +233,30 @@ The following method return types are supported:
...
 `javax.servlet.http.HttpServletRequest` and `javax.servlet.http.HttpServletResponse` are also supported.
 For this to function however, users need to add the `quarkus-undertow` dependency.
 
+=== Exception handler method return types
+
+The following method return types are supported:
+
+* `org.springframework.http.ResponseEntity`
+* `java.util.Map`
+* `void`, but in Quarkus, it must be combined with a `@ResponseStatus` annotation on the method to send an status error to HTTP response.
```
Comment (gsmet) [inline docs/src/main/asciidoc/spring-web.adoc]: @geoand I think this is not true anymore?

File: docs/src/main/asciidoc/spring-web.adoc [hunk 3]
```diff
@@ -233,12 +233,30 @@ The following method return types are supported:
...
+
+* `org.springframework.http.ResponseEntity`
+* `java.util.Map`
+* `void`, but in Quarkus, it must be combined with a `@ResponseStatus` annotation on the method to send an status error to HTTP response. 
+
+Other return types mentioned in the Spring `https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/bind/annotation/ExceptionHandler.html[ExceptionHandler javadoc]` are not supported.
+
+=== Exception handler method parameter types
+
+The following parameter types are supportted, in arbitrary order:
```
Comment (gsmet) [inline docs/src/main/asciidoc/spring-web.adoc]:
```suggestion
The following parameter types are supported, in arbitrary order:
```

File: docs/src/main/asciidoc/spring-web.adoc [hunk 4]
```diff
@@ -233,12 +233,30 @@ The following method return types are supported:
...
+
+* An exception argument: declared as a general Exception or as a more specific exception. This also serves as a mapping hint if the annotation itself does not narrow the exception types through its `value()`.
```
Comment (gsmet) [inline docs/src/main/asciidoc/spring-web.adoc]:
```suggestion
* An exception argument: declared as a general `Exception` or as a more specific exception. This also serves as a mapping hint if the annotation itself does not narrow the exception types through its `value()`.
```

File: docs/src/main/asciidoc/spring-web.adoc [hunk 5]
```diff
@@ -233,12 +233,30 @@ The following method return types are supported:
...
+* Request and/or response objects (typically from the Servlet API). You may choose any specific request/response type, e.g. `ServletRequest` / `HttpServletRequest`. To use Servlet API, you should add `qarkus-undertow` dependency.
```
Comment (gsmet) [inline docs/src/main/asciidoc/spring-web.adoc]:
```suggestion
* Request and/or response objects (typically from the Servlet API). You may choose any specific request/response type, e.g. `ServletRequest` / `HttpServletRequest`. To use Servlet API, you should add `quarkus-undertow` dependency.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
This is not ready yet. I added a couple of comments and questions.

File: docs/src/main/asciidoc/spring-web.adoc (hunk 2 above)
Comment (geoand) [reply docs/src/main/asciidoc/spring-web.adoc]: Correct, this restriction has been lifted

**Comment by geoand** [comment]
@gsmet I force pushed to the PR to address the issues

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
I checked it manually to avoid having to wait for CI. Will backport now.

**Comment by geoand** [comment]
Thanks @gsmet once again!!

File: docs/src/main/asciidoc/spring-web.adoc (hunk 2 above)
Comment (hantsy) [reply docs/src/main/asciidoc/spring-web.adoc]: I notice the `void` item has been removed. If the `void` return type is supported, I think we need a `void` in the list. I know the issue is fixed.

**Pull request merged**